app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -51,6 +56,29 @@ export const snapKeyringBuilder = (
         const addresses = await getKeyringController().getAccounts();
         return addresses.includes(address.toLowerCase());
       },
+      redirectUser: async (snapId: string, url: string, message: string) => {
+        // Either url or message must be defined
+        if (url.length > 0 || message.length > 0) {
```

Comment:
There are cases where there are no redirect URLs but there are instructions offered on how to complete the signing. Examples of this are hardware wallet snaps are the silent laboratories because they use a companion app.

Reply:
Okay, one nit then, perhaps in that case the copy shouldn't be "cancel"? Seems confusing if the message is actually useful? Perhaps it should just be OK?